src/Components/Shared/src/DefaultAntiforgeryStateProvider.cs

```diff
@@ -25,7 +26,7 @@ public DefaultAntiforgeryStateProvider(PersistentComponentState state)
         {
             state.PersistAsJson(PersistenceKey, _currentToken);
             return Task.CompletedTask;
-        });
+        }, new InteractiveAutoRenderMode());
```

Comment:
Does this mean we're going to emit prerendered state for WebAssembly even if you're not using WebAssembly? In fact, even if you're not using any interactive rendering at all?

Reply:
No, it's covered in a later commit, but essentially if you only do `AddWebAssemblyRenderMode` or `AddServerRenderMode` we only emit the comment for webassembly or server respectively. 

If you have declared that you are using both, then we will emit it. I haven't done it, but we can look at the set of actual render modes that a component used during a request and narrow down to a single render mode if possible too, in that situation.

The point here, is that auto means, "wherever it's actually needed" as opposed to all locations.